src/main/resources/templates/db_scripts/migration/extract_credits_from_expenditures.sql

```diff
@@ -45,14 +40,11 @@ WITH aggregated_amounts AS (
 )
 UPDATE ${myuniversity}_${mymodule}.transaction AS encumbrance
 SET
```

Comment:
Can you please describe why this difference affects locking?

Reply:
my understanding is how it will insert to json old one set specific field and will keep existing content including previous version, jsonb || jsonb_build_object is merge to existing with not affecting existing struture. can't find much info from internet from gpt
Optimistic Locking Safety:
The jsonb || operator preserves all existing fields including _version
jsonb_set only updates specific paths, which is more rigid

Reply:
Did you test it also in isolation inside DBeaver/PgAdmin, if `jsonb || jsonb_build_object()` does allow to just update a single field without any locking issues?

Reply:
One final check, are the result values correct/what you expect?

Reply:
yes values are correct, main changes is how extra json object modifying not logic changes. but when testing some issue happening https://github.com/folio-org/mod-finance-storage/pull/434#discussion_r1852179499 working on to fix